### PR TITLE
Add 'getSubscriptions' API on Server.

### PR DESCRIPTION
**Pull request by teranisi**
Here is an implementation of [#207](https://github.com/andsel/moquette/issues/207) and related test.

### CODE REVIEW

File: broker/src/test/java/io/moquette/server/ServerIntegrationEmbeddedPublishTest.java [hunk 1]
```diff
@@ -97,6 +97,17 @@ private void internalPublishToWithQosAndRetained(String topic, QOSType qos, bool
         m_server.internalPublish(message);
     }
 
+    private void connectNoCleanSession() throws Exception {
+        MqttConnectOptions opts = new MqttConnectOptions();
+        opts.setCleanSession(false);
+        m_subscriber.connect(opts);
+    }
+
+    private void subscribeToWithQosAndNoCleanSession(String topic, int qos) throws Exception {
```
Comment (kevto) [inline broker/src/test/java/io/moquette/server/ServerIntegrationEmbeddedPublishTest.java:106]: Unit test is not verifying any constraints or whatsoever.

File: broker/src/test/java/io/moquette/server/ServerIntegrationEmbeddedPublishTest.java [hunk 2]
```diff
@@ -97,6 +97,17 @@ private void internalPublishToWithQosAndRetained(String topic, QOSType qos, bool
         m_server.internalPublish(message);
     }
 
+    private void connectNoCleanSession() throws Exception {
```
Comment (kevto) [inline broker/src/test/java/io/moquette/server/ServerIntegrationEmbeddedPublishTest.java:100]: Unit test is not verifying any constraints or whatsoever.

**Review by kevto — CHANGES_REQUESTED** [review]
Two notes regarding unit tests.

**Comment by kevto** [comment]
Perhaps also an idea to make these `List` instances immutable? I don't know what your exact intentions are but if you're only looking for to read data and not mutate any of it, to make these objects immutable rather than allowing out of package classes to modify `Subscription` data.

**Comment by andsel** [comment]
I agree with @kevto no mutable state has to be shared. I'll review the pr
when I'm back from holiday
Andre
Il giorno 16/set/2016 16:19, "Kevin" notifications@github.com ha scritto:

**Comment by teranisi** [comment]
My intention was to provide a read interface for an external program. The objects can be immutable.

**Pull request merged**